### PR TITLE
Phaser renderer region

### PR DESCRIPTION
**Pull request by Grenagar**

### CODE REVIEW

File: ts/src/renderer/phaser/classes/PhaserRegion.ts [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+		if (region._destroyed || container._destroyed) {
+			this.scene.events.off('update', this.update, this);
+			this.destroy();
+			return;
+		}
+
+		const stats = this.region._stats.default
+
+		if (this.x != stats.x) this.x = stats.x;
```
Comment (pavle-goloskokovic) [inline ts/src/renderer/phaser/classes/PhaserRegion.ts:48]: these checks are unnecessary as coordinates assignment can happen on each frame. doing two additional check will only affect performance

File: ts/src/renderer/phaser/classes/PhaserRegion.ts [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+		if (this.y != stats.y) this.y = stats.y;
+
+		if (this.width != stats.width || this.height != stats.height) {
```
Comment (pavle-goloskokovic) [inline ts/src/renderer/phaser/classes/PhaserRegion.ts:51]: please use `!==` and `===` for comparisons always

**Comment by Grenagar** [comment]
--game=62717c3ec749f41d7199aed2 - game id where we can see regions rendering

File: ts/src/renderer/phaser/classes/PhaserRegion.ts [hunk 3]
```diff
@@ -0,0 +1,62 @@
+class PhaserRegion extends Phaser.GameObjects.Graphics {
+
+	private width;
```
Comment (pavle-goloskokovic) [inline ts/src/renderer/phaser/classes/PhaserRegion.ts:3]: define types for these properties

**Review by pavle-goloskokovic — APPROVED** [review]

**Review by sloont — APPROVED** [review]

**Pull request merged**